src/hardware/pcspeaker_discrete.cpp

```diff
@@ -362,12 +359,12 @@ void PcSpeakerDiscrete::ChannelCallback(const uint16_t frames)
 	float buf[render_frames];
 
 	ForwardPIT(1);
-	last_index            = 0.0f;
-	uint16_t pos          = 0u;
-	auto sample_base      = 0.0f;
+	last_index       = 0.0f;
+	auto sample_base = 0.0f;
+
+	bool enteries_were_dequeued = false;
 
-	const auto period_per_frame_ms = 1.0f / frames - FLT_EPSILON;
-	// The epsilon reduction ensures the sum of the periods is <= 1ms
+	const auto period_per_frame_ms = 1.0001f / frames;
```

Comment:
Why not just 1.0f?

Reply:
Deja vu: https://github.com/dosbox-staging/dosbox-staging/pull/1690#discussion_r902189308

This factor is now better understood, and I've described it as a comment in the code.